extensions/interactions/RatioExpressionInput/directives/ratio-expression-input-validation.service.ts

```diff
@@ -0,0 +1,192 @@
+// Copyright 2020 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Validator service for the RatioExpressionInput interaction.
+ */
+
+import { Injectable } from '@angular/core';
+import { downgradeInjectable } from '@angular/upgrade/static';
+
+import { AnswerGroup } from
+  'domain/exploration/AnswerGroupObjectFactory';
+import { Warning, baseInteractionValidationService } from
+  'interactions/base-interaction-validation.service';
+import { RatioExpressionInputCustomizationArgs } from
+  'extensions/interactions/customization-args-defs';
+import { RatioObjectFactory, Ratio } from
+  'domain/objects/RatioObjectFactory';
+import { RatioExpressionInputRulesService } from
+  './ratio-expression-input-rules.service';
+import { Outcome } from
+  'domain/exploration/OutcomeObjectFactory';
+import { AppConstants } from 'app.constants';
+
+
+@Injectable({
+  providedIn: 'root'
+})
+export class RatioExpressionInputValidationService {
+  constructor(
+    private rof: RatioObjectFactory,
+    private baseInteractionValidationServiceInstance:
+      baseInteractionValidationService) {}
+
+  getCustomizationArgsWarnings(
+      customizationArgs: RatioExpressionInputCustomizationArgs): Warning[] {
+    var isInt = function(n) {
+      return angular.isNumber(n) && n % 1 === 0;
+    };
+    var minimumNumberOfTerms =
+      customizationArgs.numberOfTerms.value;
+    if (!isInt(minimumNumberOfTerms)) {
+      return [
+        {
+          type: AppConstants.WARNING_TYPES.ERROR,
+          message: (
+            'Number of terms must be integral.')
+        }
```

Comment:
A couple of minor things to note:
 - This warning gets shown even when it's not really relevant, for eg, when the input field is empty.
 - You may want to change the isInt checker -- I entered a decimal and it got rounded off to an integer

![nits](https://user-images.githubusercontent.com/35144226/91036609-44c28a80-e625-11ea-9842-0a6ec57348d4.gif)

Reply:
why would someone would want to enter such value in the first place and it does get's rounded off because that's done by javascript itself and it cannot store that value so it would just round of that value which i think is completely fine.
@BenHenning @seanlip your thoughts ?
.

Reply:
Thanks @iamprayush for checking this.

@aggarwalpulkit596 could we guard against this by also ensuring the input doesn't contain a decimal at all? That might be a more reliable check than the mod check currently being done.

Also, shouldn't we ensure the number of terms are also > 0?

Reply:
We could do that but instead of doing the above displaying error message is much more sensible to me as we can customise this message as per needs instead of just not allowing the input in the first place and yes we are ensuring of a positive number here and this same approach is being used for the text input interaction number of rows as well so i think we should stick to this only.

Reply:
@aggarwalpulkit596 I've read your last comment three times and I still can't figure out what it is saying. Can you please use shorter sentences and be clear about what approach you want to pursue (precisely) and why?

Reply:
I'm just saying that instead of not able to type decimal values we should be throwing an error because that way we could customize the message that would be shown to user.
If we disable it right away creator might think why he's not able to type decimal.

Reply:
@seanlip does that clarifies ?

Reply:
Yes, that sounds fine to me. Just make sure that the submit button is properly disabled (it should not be green and clickable) if the decimal is invalid.

Reply:
That's already being taken care of.

Reply:
@seanlip should I change the warning message in the case of empty value?

Reply:
Yup, always be specific. Thanks!

Reply:
@iamprayush done

Reply:
@aggarwalpulkit596 This is still not done. If invalid inputs like '--3' or '++3' are entered, the warning reads "Number of terms cannot be empty." which is not right.
Also, if a negative number is entered, no warnings are shown, only the save button is disabled and the input box turns red.

Reply:
We could handle negative values but handling something like expression is not feasible because an empty value and a value like `--1`  both are treated as undefined because we are having the type as a number so we could a common message for both @seanlip @iamprayush any suggestion what should be the warning message?

Reply:
@seanlip @iamprayush just to clarify the above comment
In the backend, for numberOfTerms customization arg we type as number and if anything other than number is given as input - e.b --1--1 or -1-1-1-1 or any expression it would basically convert it to undefined and therefore we can't have different warnings for empty and invalid values(empty value is undefined too).
I was thinking to show a common warning if the value is defined and we are already handling negative and decimal values separately.
Error message 
Undefined Value(Empty Value too) -  Number of terms should be an integer value.
Decimal Value -  Number of terms cannot have decimal places.
Negative Value -  Number of terms must be a positive integer.

Reply:
Thanks for the clear explanation, @aggarwalpulkit596! The error messages you've listed above seem fine to me.

Reply:
Ah, I see. Then let's just go with a standard "The number of terms should be a positive integer." for all the cases.

Reply:
@iamprayush updated